pkg/metrics/ddl.go

```diff
@@ -180,26 +180,48 @@ func InitDDLMetrics() {
 const (
 	LblAction = "action"
 
-	LblAddIndex      = "add_index"
-	LblAddIndexMerge = "add_index_merge_tmp"
-	LblModifyColumn  = "modify_column"
-
+	// Used by BackfillProgressGauge
+	LblAddIndex       = "add_index"
+	LblAddIndexMerge  = "add_index_merge_tmp"
+	LblModifyColumn   = "modify_column"
 	LblReorgPartition = "reorganize_partition"
+
+	// Used by BackfillTotalCounter
+	LblAddIdxRate         = "add_idx_rate"
+	LblMergeTmpIdxRate    = "merge_tmp_idx_rate"
+	LblCleanupIdxRate     = "cleanup_idx_rate"
+	LblUpdateColRate      = "update_col_rate"
+	LblReorgPartitionRate = "reorg_partition_rate"
 )
 
-// GenerateReorgLabel returns the label with schema name and table name.
-func GenerateReorgLabel(label string, schemaName string, tableName string) string {
+// generateReorgLabel returns the label with schema name, table name and optional column/index names.
+// Multiple columns/indexes can be concatenated with "+".
+func generateReorgLabel(label, schemaName, tableName, colOrIdxNames string) string {
 	var stringBuilder strings.Builder
-	stringBuilder.Grow(len(label) + len(schemaName) + len(tableName) + 2)
+	if len(colOrIdxNames) == 0 {
+		stringBuilder.Grow(len(label) + len(schemaName) + len(tableName) + 6)
+	} else {
+		stringBuilder.Grow(len(label) + len(schemaName) + len(tableName) + len(colOrIdxNames) + 9)
+	}
 	stringBuilder.WriteString(label)
-	stringBuilder.WriteString("_")
+	stringBuilder.WriteString("_[")
```

Comment:
Is there another usage for "[" in labels?

Reply:
Not found. Maybe `dbname-tablename-col/idxname` is a better way. Let me modify it

Reply:
Fixed in https://github.com/pingcap/tidb/pull/58380/commits/5b8a1117004ce9aba4dfebfc1b53f63a021cc97f, PTAL